### PR TITLE
Ensure all of the task labels can be parsed in the task graph

### PR DESCRIPTION
**Pull request by gregtatum**
This is a break out from #611 and #606. It reworks the task label parsing, renames the mechanism to use "task label" instead of "tag", and includes other fixes and cleanups to the mechanisms. The commits should be logically ordered for the work.

### CODE REVIEW

File: tests/test_tracking_utils.py [hunk 1]
```diff
@@ -58,10 +59,42 @@
...
-def test_parse_tag(example, parsed_values):
-    assert parse_tag(example) == parsed_values
+def test_parse_task_label(task_label, parsed_values):
+    assert parse_task_label(task_label) == ParsedTaskLabel(*parsed_values)
+
+
+def test_parse_labels_on_full_taskgraph():
+    """Ensure that all the taskgraph task labels parse."""
+    for task in get_full_taskgraph():
+        if not task.startswith("train-") and not task.startswith("evaluate-"):
```
Comment (eu9ene) [inline tests/test_tracking_utils.py:88]: nice, we'll catch all the issues in the tests now. Why are you skipping train- and evaluate-? Can we just enable this test for everything?

File: tracking/translations_parser/utils.py [hunk 2]
```diff
@@ -12,49 +13,104 @@
...
+    #
+    # Capture any augmentations
+    #   evaluate-teacher-flores-flores_aug-title_devtest-lt-en-1_2
+    #                                  ^^^^^^^^^
     r"(?P<aug>aug-[^_]+)?"
-    r"_?(?P<dataset>[-\w_]*?(-[a-z]{3}-[a-z]{3})?)?"
-    r"-?(?P<lang>[a-z]{2}-[a-z]{2})?"
+    #
+    # Capture the dataset.
+    #   evaluate-quantized-mtdata_aug-mix_Neulab-tedtalks_eng-lit-lt-en
```
Comment (eu9ene) [inline tracking/translations_parser/utils.py:79]: nice docs!

**Review by eu9ene — APPROVED** [review]
lgtm, let's unblock other PRs but we should wait for CI to pass. FYI @La0 @vrigal

File: tests/test_tracking_utils.py (hunk 1 above)
Comment (gregtatum) [reply tests/test_tracking_utils.py:88]: The logic is inverted here, so if it's not train and not evaluate, it's skipping. Maybe I should rewrite it as:

`if not (task.startswith("train-") or task.startswith("evaluate-")):`

**Pull request merged**